app/services/hyrax/collection_types/create_service.rb

```diff
@@ -40,6 +42,20 @@ def self.create_collection_type(machine_id: DEFAULT_MACHINE_ID, title: DEFAULT_T
           c.assigns_workflow = opts[:assigns_workflow]
           c.assigns_visibility = opts[:assigns_visibility]
         end
+        add_participants(ct.id, opts[:participants])
+        ct
+      end
+
+      # @param collection_type_id [Integer]
+      # @param participants [Array]
+      def self.add_participants(collection_type_id, participants)
+        return unless collection_type_id && participants.count > 0
+        participants.each do |p|
+          agent_type = p[:agent_type]
```

Comment:
If the given participant is `{ }`, we end up with `agent_type = nil` is that an acceptable value for the `.create` I'm wondering about using `p.fetch(:agent_type)` instead. (As well as for `p.fetch(:agent_id)` and `p.fetch(:access)`).

Reply:
I would prefer to see add_participants as part of the [PermissionsService](https://github.com/samvera/hyrax/blob/collections-sprint/app/services/hyrax/collection_types/permissions_service.rb) since that service deals with managing and querying participants.

Reply:
RE: acceptable values
* agent_type must be either `'group'` or `'user'`
* access must be either `'manage'` or `'create'`
* none of the values can be nil

Validation should be added to the model for that.

Reply:
* Updated to use fetch.  Model already has `presence` validation of all fields.
* Created an issue to cover validation isssues..  #1649 Collection Types: Add validations to participants model 
* Method was moved to PermissionsService